src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs

```diff
@@ -246,6 +246,14 @@ public partial class FixAllContext
         /// </summary>
         public FixAllContext WithCancellationToken(CancellationToken cancellationToken)
         {
+            // TODO: We should change this API to be a virtual method, as the class is not sealed.
+            //       For now, special case FixMultipleContext.
+            var fixMultipleContext = this as FixMultipleContext;
+            if (fixMultipleContext != null)
+            {
+                return fixMultipleContext.WithCancellationToken(cancellationToken);
```

Comment:
I thought there is type hierarchy. shouldn't this be virtual and then overriden rather than base type (which I think is reversed) knows about derived type and then have "if" statement.